### ISSUE TITLE
Error with timezones

### ISSUE THREAD

**Issue by joniw** [LukasHH/mod_itcs_reservation_calendar#1]
Hello,

after upgrading to the newest version the reservation calendar on our homepage was not working anymore. After some digging it seems to me, that there is an error in the function used before painting the days of the calendar:
https://github.com/LukasHH/mod_itcs_reservation_calendar/blob/328dbe276e5c930dee7f9c6b3d53f07758d270d8/tmpl/default.php#L56

The function gets the day in the timezone defined by the browser, the dates in the datebase seem to be utc. To fix the offset you can use `date.setMinutes(date.getMinutes() - date.getTimezoneOffset());` as the first line of the function:

```javascript
function (date){
    date.setMinutes(date.getMinutes() - date.getTimezoneOffset()); // fix offset from utc
    var chkDate = Math.round(date.getTime()/1000); // get timestamp
    var r ='';
    var c ='';

    disableDays_<?php echo 'id'.$uniqid; ?>.forEach(disableDay => {
        if(disableDay.t == chkDate){
            r = disableDay.cal_day_info;
            c = 'reserved tt ' + disableDay.cal_day_color;
        }
    });
    return {tooltip: r, classes: c};
}
```

I am not sure if this works in general so please have a look at this problem.

**Comment by LukasHH** [comment]
The problem is the stored date in the database. In the new version, the timestamps are compared on the basis of the 0 o'clock date. I did this because the time zones always led to problems with the day change.

With `date.setMinutes(date.getMinutes() - date.getTimezoneOffset());` you change the timestamp of the days from the calendar. This means that the timestamps of the calendar are no longer correct.

I think you have to take the time zone of the server into account when reading out the saved data and also specify server_utc as a filter in the calendar field.

**Comment by LukasHH** [comment]
I have now tested again with different time zone settings and can't find any problems.

What was the previous version, before the update?
If it was version 1.1.7 or older, clear the browser cache once.

**Comment by joniw** [comment]
Thank you for looking at the issue.

I did an update of joomla itself and the underlying container. I also tried opening the site with an incognito tab, even another browser, the cache should not be a problem.

I use the [bitnami joomla container](https://hub.docker.com/r/bitnami/joomla) version 3.6.10. It uses the utc timezone as default. When i set the timezone of the container to my browser timezone everything is working as expected.

So to me it looks like the date given to `beforeShowDay` contains the timezone of the browser, which leads to an offset. The problem is also described [here](https://stackoverflow.com/questions/22717201/how-to-avoid-time-zone-issues-with-jquery-datepicker).

@Edit: When using a different timezone in the container then utc, my fix is not working because the dates from the database are loaded with this offset

**Comment by joniw** [comment]
The linked pull request fixes the problem for me. I have tested it with mutliple client and server timezone settings

**Comment by LukasHH** [comment]
I still cannot reproduce the problem. I do not use the container, but the classic installation of Joomla. Tested both variants locally and on my server with different time zones.

According to my tests, it unfortunately makes the fix unusable, because it then only works with UTC. However, it is better to set the correct time zone in the Joomla configuration.

Without the fix, it also works for me if UTC is set in the Joomla configuration and a different time zone locally.

So far, it seems that you are the only one with the problem. Otherwise, many others also contact me if there is a problem.

In this context, however, I have found something else. What can lead to a problem if a different time zone is set in your own Joomla profile than in the configuration of Joomla.
